arduino/smartcar/smartcar.ino

```diff
@@ -1,83 +1,200 @@
-#include <Smartcar.h>
-#include <vector>
 #include <MQTT.h>
+#include <Smartcar.h>
 #include <WiFi.h>
 
+#include <vector>
+#ifdef __SMCE__
+#include <OV767X.h>
+#endif
+
 MQTTClient mqtt;
 WiFiClient net;
 
-//wifi SSID and password defined below
-//empty ssid and pass means connect to localhost (only change for physical car)
+// wifi SSID and password defined below
+// empty ssid and pass means connect to localhost (only change for physical car)
 const char ssid[] = " ";
 const char pass[] = " ";
 
-//Identifying tag
+// Identifying tag
 const String ID_TAG = "MQTTSmartCar";
-//main topic to derive others from
+// main topic to derive others from
 const String MAINMQTT_TOPIC = "/smartcar/";
-//control topics
-const String STEERING_TOPIC = MAINMQTT_TOPIC + "controls/steering"; 
-const String THROTTLE_TOPIC = MAINMQTT_TOPIC + "controls/throttle"; 
+// control topics
+const String STEERING_TOPIC = MAINMQTT_TOPIC + "controls/steering";
+const String THROTTLE_TOPIC = MAINMQTT_TOPIC + "controls/throttle";
 const String ESTOP_TOPIC = MAINMQTT_TOPIC + "controls/stop";
 
-//sensor topics
+// sensor topics
 const String IR_TOPIC = MAINMQTT_TOPIC + "sensor/ir";
 const String ULTRASONIC_TOPIC = MAINMQTT_TOPIC + "sensor/ultrasonic";
 const String GYROSCOPE_TOPIC = MAINMQTT_TOPIC + "sensor/gyroscope";
 const String ODOMETER_TOPIC = MAINMQTT_TOPIC + "sensor/odometer";
-const String CAMERA_TOPIC = MAINMQTT_TOPIC + "sensor/camera";
-//status topics
+const char CAMERA_TOPIC[] = "/smartcar/sensor/camera";
+// status topics
 const String BLINKERS_TOPIC = MAINMQTT_TOPIC + "status/blinkers";
 
-String speed = "-1.0";
+// calculated data
+const String ODOMETER_DISTANCE = MAINMQTT_TOPIC + "status/odometer/distance";
+const String ODOMETER_SPEED = MAINMQTT_TOPIC + "status/odometer/speed";
+
+const int MAX_DISTANCE = 80;
+const int TRIGGER_PIN = 6;
+const int ECHO_PIN = 7;
+
+const unsigned long PULSES_PER_M = 400;
+
+String blinkerStatus = "off";
+
+ArduinoRuntime arduinoRuntime;
+// Motors
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(rightMotor, leftMotor);
+SimpleCar car(control);
+
+// Sensors
+SR04 frontUltrasonic(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+GP2Y0A21 frontIR(arduinoRuntime, 0);
+GY50 gyroscope(arduinoRuntime, 0, 100UL);
+DirectionlessOdometer leftOdometer(
+    arduinoRuntime, smartcarlib::pins::v2::leftOdometerPin,
+    []() { leftOdometer.update(); }, PULSES_PER_M);
+DirectionlessOdometer rightOdometer(
+    arduinoRuntime, smartcarlib::pins::v2::rightOdometerPin,
+    []() { leftOdometer.update(); }, PULSES_PER_M);
```

Comment:
Why aren't you creating a `SmartCar` but instead using a `SimpleCar` with sensors on the side?